doc/source/whatsnew/v0.18.1.txt

```diff
@@ -22,13 +22,62 @@ New features
 Enhancements
 ~~~~~~~~~~~~
 
+Method chaininng improvements
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Following methods / ``Indexer`` now accept ``callable`` as input. It is
+intended to make these more useful in the method chain.
+(:issue:`11485`, :issue:`12533`)
+
+- ``.where`` and ``.mask``
+- ``.query``
+- ``.loc``, ``iloc`` and ``.ix``
+
+When ``callable`` is passed, it is evaluated on the caller
+``Series/DataFrame`` prior to any other logics.
+
+.. note::
+
+   Thus, if input is callable ``np.ndarray`` or ``Series/DataFrame`` class
+   which defined on your own, it is evaluated on the caller
+   ``Series/DataFrame``, not regarded as values.
+
+``.where`` and ``.mask`` can now accept callable as condition and other
+arguments.
+
+.. ipython:: python
+
+   df = pd.DataFrame({'A': [1, 2, 3],
+                      'B': [4, 5, 6],
+                      'C': [7, 8, 9]})
+   df.where(lambda x: x > 4, lambda x: x + 10)
+
+``.query`` can now accept callable as expr argument.
+
+.. ipython:: python
+
+   df.query(lambda x: x.A >= 2)
+
+``.loc``, ``.iloc`` and ``.ix`` can now accept callable, and tuple of callable
+as a slicer. The callable can return valid ``bool`` indexer or anything which
+is valid for these indexer's input.
+
+In most use case are covered by callable returns ``bool``
+indexer as below.
+
+.. ipython:: python
+
+   # callable returns bool indexer
```

Comment:
need to have some tests if people try to pass something like:

``df.loc[lambda x: ...., list_of_columns]`` which is a bit odd, but I suppose ok, otherwise we need to interpret this